src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -180,6 +181,26 @@ private boolean deleteResourceInner(AuthenticatedUserRequest userReq,
         return policies;
     }
 
+    @Override
+    public Map<IamRole, String> getDatasetPolicyEmails(AuthenticatedUserRequest userReq, UUID datasetId)
+        throws InterruptedException {
+        SamRetry samRetry = new SamRetry(configurationService);
+        return samRetry.perform(() -> getDatasetPolicyEmailsInner(userReq, datasetId));
+    }
+
+    private Map<IamRole, String> getDatasetPolicyEmailsInner(AuthenticatedUserRequest userReq, UUID datasetId)
+        throws ApiException {
+
+        GoogleApi samGoogleApi = samGoogleApi(userReq.getRequiredToken());
+        Map<IamRole, String> policies = new HashMap<>();
+        for (IamRole role : Arrays.asList(IamRole.STEWARD, IamRole.CUSTODIAN, IamRole.INGESTER)) {
+            SyncStatus status =
+                samGoogleApi.syncStatus(IamResourceType.DATASET.name(), datasetId.toString(), role.name());
```

Comment:
I don't think you want to sync the policies at this point. The policies do not need to be updated.
All you need are the group names. Maybe we need to save those from when we create the dataset.

Reply:
I had considered that as well, I wasn't sure if there'd be a chance that Sam switched up the groups (I guess that's unlikely since they don't have a mechanism to update the groups at every use-site).

The naming of the method here is a bit misleading though; `syncStatus` is fetching the latest status of synchronization (GET), not triggering a sync of statuses (POST).

Reply:
My main concern for the track-policies-in-the-DB case is figuring out how we migrate existing datasets.

Reply:
Hmmm but even assuming we still ask Sam for the group emails each time, this should probably be broken into multiple steps for the bulk-load case:
1. Step that gets the policy emails from somewhere and sticks them in the flight map
2. Step that sets the ACLs for the policy emails